homework/myworld/web_scrapper.py

```diff
@@ -0,0 +1,91 @@
+import psycopg2
+import requests
+import re
+from bs4 import BeautifulSoup, element
+import os
+        
+# For the credentials mentioned below, you may refer the docker-compose.yml present in myworld .
+db_name = 'employee_db'
+db_user = 'postgres'
+db_pass = '123456'
+db_host = 'psql-db2'
+db_port = '5432'
+        
+# This will create the connection the to postgres database.
+conn = psycopg2.connect(dbname=db_name, user=db_user, password=db_pass, host=db_host, port=db_port)
+        
+        
+def add_row_to_blog(title, author, date, time,content,recommended,path):
+    # This function will add the entry to database
+    sql = """INSERT INTO members_blog (title, release_date, blog_time, author, created_date, content, recommended, path) VALUES (%s, %s::DATE, %s::TIME, %s, NOW(),%s,%s,%s)"""
+            
+    with conn:
+        with conn.cursor() as curs:
+            time = time.replace('\u202f', "")
+            curs.execute(sql, (title, date, time, author, content, recommended, path))
+        
+        
+def truncate_table():
+    # This function will delete the existing entries from the database.
+    with conn:
+        with conn.cursor() as curs:
+            curs.execute("TRUNCATE members_blog CASCADE;")
+        
+        
+def start_extraction():
+    print("Extraction started")
+    url = "https://blog.python.org/"
+            
+    # Each time when we add new entry we delete the existing entries.
+    truncate_table()
+    data = requests.get(url)
+    page_soup = BeautifulSoup(data.text, 'html.parser')
+            
+    # Getting all the articles
+    blogs = page_soup.select('div.date-outer')
+    
+    fp=open("scrap_data.html","w")
+    path=os.path.abspath("scrap_data.html")
+            
+    for blog in blogs:
```

Comment:
@rajashree-k If any exception occurs in this loop, what happens to the file?

Reply:
I think file will create, but if any exception occurs in the loop that time it might stops execution of the next line. Because here not used any exception handling like with, try & catch..

Reply:
@rajashree-k it's obvious that the moment your code encounters a exception it would break and stop the execution of code. But the question still stands, what happens to the opened file?